src/data/equiv/tree.lean

```diff
@@ -0,0 +1,138 @@
+
+/-
+Copyright (c) 2019 E.W.Ayers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: E.W.Ayers.
+
+Prove that branching recursive datastructures are encodable by writing a little verified lisp parser.
```

Comment:
Ed, recently we have been trying to improve our documentation practices. In particular, we want to use "module docstrings" at the top of each file for the "overview comment". The syntax is
```
/-!
# Header

Some text
-/
```

Reply:
Could you point me to an example of module docstrings?

Reply:
In short, the text about the file which used to go in the same comment as the copyright notice now goes in between `/-!` and `-/` after the imports.

See the top of `meta/expr.lean` in #1723; see also the "Header comment" section of the "contribute" doc file edited in that PR.

Reply:
https://github.com/leanprover-community/mathlib/blob/master/docs/contribute/doc.md describes the documentation guidelines we added this summer.